daemon/logger/awslogs/cloudwatchlogs.go

```diff
@@ -270,14 +277,17 @@ func (l *logStream) collectBatch() {
 // publishBatch calls PutLogEvents for a given set of InputLogEvents,
 // accounting for sequencing requirements (each request must reference the
 // sequence token returned by the previous request).
-func (l *logStream) publishBatch(events []*cloudwatchlogs.InputLogEvent) {
+func (l *logStream) publishBatch(events []wrappedEvent) {
 	if len(events) == 0 {
 		return
 	}
 
+	// events in a batch must be sorted by timestamp
+	// see http://docs.aws.amazon.com/AmazonCloudWatchLogs/latest/APIReference/API_PutLogEvents.html
 	sort.Sort(byTimestamp(events))
+	cwEvents := getCWEvents(events)
```

Comment:
The [`PutLogEvents` API](http://docs.aws.amazon.com/AmazonCloudWatchLogs/latest/APIReference/API_PutLogEvents.html) has this requirement:

> The log events in the batch must be in chronological ordered by their `timestamp`.

The sorting here is to ensure that even on a system that is experiencing some changes in the clock (e.g., skew being corrected by NTP) that the call to `PutLogEvents` will still succeed.